lib/std/testing.zig

```diff
@@ -6,25 +6,36 @@ const print = std.debug.print;
 
 pub const FailingAllocator = @import("testing/failing_allocator.zig").FailingAllocator;
 
-/// This should only be used in temporary test programs.
-pub const allocator = allocator_instance.allocator();
-pub var allocator_instance = b: {
+fn checkScope() void {
```

Comment:
QoL suggestion: have this function take a `std.builtin.SourceLocation` as a parameter and do:
```
@compileError(std.fmt.comptimePrint("{s} can only be used inside `test` blocks", .{src.fn_name}));
```

Reply:
Ah, nice one. I was thinking how we could improve this error and this is already a lot better.

Reply:
However, when I do this... it just kinda doesn't work anymore? It just gives me this without any info:
```
zig$ zig build test-std --zig-lib-dir lib
error: test...
error: The following command terminated unexpectedly:
```
Don't know what's up with that.
Also this option should probably be compared to my idea above of checking `builtin.is_test` when exporting `testing` to the std. I'm imagining the stack trace would be even worse with my idea above.

Reply:
Could it be that you're calling `@src` outside of a function? Remember to test with `-fstage1` as well.

Reply:
That doesn't make a lot of sense lol, you're accessing the optional without unwrapping it first. And the parameter isn't `comptime`.

Reply:
Oh, I just realized that. Would've noticed if the compiler had given me a proper error.
Well after looking at it more closely, this should work, right?
```zig
fn checkScope(maybe_src: ?std.builtin.SourceLocation) void {
    if (!builtin.is_test) {
        if (maybe_src) |src|
            @compileError(src.fn_name ++ " can only be used inside `test` blocks")
        else
            @compileError("this can only be used inside `test` blocks");
    }
}
```
Still the same problem unfortunately...

Reply:
weird, oh well.